WordPress/Classes/Services/NotificationSupportService.swift

```diff
@@ -0,0 +1,34 @@
+import Foundation
+
+@objc
+open class NotificationSupportService: NSObject {
+    /// Sets the OAuth Token that should be used by the Notification Service Extension to access WPCOM.
+    ///
+    /// - Parameter oauth2Token: WordPress.com OAuth Token
+    ///
+    @objc
+    class func insertExtensionToken(_ oauthToken: String) {
+        do {
+            try SFHFKeychainUtils.storeUsername(WPNotificationServiceExtensionKeychainTokenKey,
+                                                andPassword: oauthToken,
+                                                forServiceName: WPNotificationServiceExtensionKeychainServiceName,
+                                                accessGroup: WPAppKeychainAccessGroup,
+                                                updateExisting: true)
+        } catch {
+            DDLogDebug("Error while saving Notification Service Extension OAuth token: \(error)")
+        }
+    }
+
+    /// Attempts to the current WPCOM OAuth Token
```

Comment:
Nitpick – `/// Attempts to *delete* the current WPCOM OAuth Token`

Reply:
Good catch! I have pushed an additional commit that addresses your feedback.